### ISSUE TITLE
Missing proxy classes needed for MSK IAM

### ISSUE THREAD

**Issue by singhbaljit** [streamthoughts/jikkou#312]
**Describe the bug**
[The MSK IAM library uses AWS SDK's Default Credentials Provider Chain](https://github.com/aws/aws-msk-iam-auth#configuring-a-kafka-client-to-use-aws-iam). When assuming another IAM role or using Web Identity Token, the AWS SDK makes HTTP calls to AWS APIs. However, due the nature of native compilation, classes needed to perform this action do not work due to usage of Reflection. Here is the error log:

```
org.apache.kafka.common.KafkaException: Failed to create new KafkaAdminClient
	at org.apache.kafka.clients.admin.KafkaAdminClient.createInternal(KafkaAdminClient.java:551)
	at org.apache.kafka.clients.admin.Admin.create(Admin.java:144)
	at org.apache.kafka.clients.admin.AdminClient.create(AdminClient.java:49)
	at io.streamthoughts.jikkou.kafka.internals.admin.DefaultAdminClientFactory.createAdminClient(DefaultAdminClientFactory.java:60)
	at io.streamthoughts.jikkou.kafka.internals.admin.AdminClientContext.getAdminClient(AdminClientContext.java:171)
	at io.streamthoughts.jikkou.kafka.health.KafkaBrokerHealthIndicator.getHealth(KafkaBrokerHealthIndicator.java:81)
	at io.streamthoughts.jikkou.client.command.health.GetHealthCommand.call(GetHealthCommand.java:100)
	at io.streamthoughts.jikkou.client.command.health.GetHealthCommand.call(GetHealthCommand.java:40)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at io.streamthoughts.jikkou.client.Jikkou$1.execute(Jikkou.java:153)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.streamthoughts.jikkou.client.Jikkou.execute(Jikkou.java:136)
	at io.streamthoughts.jikkou.client.Jikkou.main(Jikkou.java:119)
Caused by: org.apache.kafka.common.KafkaException: com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface org.apache.http.conn.HttpClientConnectionManager, interface org.apache.http.pool.ConnPoolControl, interface com.amazonaws.http.conn.Wrapped] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
	at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:184)
	at org.apache.kafka.common.network.ChannelBuilders.create(ChannelBuilders.java:192)
	at org.apache.kafka.common.network.ChannelBuilders.clientChannelBuilder(ChannelBuilders.java:81)
	at org.apache.kafka.clients.ClientUtils.createChannelBuilder(ClientUtils.java:105)
	at org.apache.kafka.clients.admin.KafkaAdminClient.createInternal(KafkaAdminClient.java:522)
	... 18 more
Caused by: com.oracle.svm.core.jdk.UnsupportedFeatureError: Proxy class defined by interfaces [interface org.apache.http.conn.HttpClientConnectionManager, interface org.apache.http.pool.ConnPoolControl, interface com.amazonaws.http.conn.Wrapped] not found. Generating proxy classes at runtime is not supported. Proxy classes need to be defined at image build time by specifying the list of interfaces that they implement. To define proxy classes use -H:DynamicProxyConfigurationFiles=<comma-separated-config-files> and -H:DynamicProxyConfigurationResources=<comma-separated-config-resources> options.
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.util.VMError.unsupportedFeature(VMError.java:89)
	at org.graalvm.nativeimage.builder/com.oracle.svm.core.reflect.proxy.DynamicProxySupport.getProxyClass(DynamicProxySupport.java:171)
	at java.base@17.0.8/java.lang.reflect.Proxy.getProxyConstructor(Proxy.java:47)
	at java.base@17.0.8/java.lang.reflect.Proxy.newProxyInstance(Proxy.java:1037)
	at com.amazonaws.http.conn.ClientConnectionManagerFactory.wrap(ClientConnectionManagerFactory.java:54)
	at com.amazonaws.http.apache.client.impl.ApacheHttpClientFactory.create(ApacheHttpClientFactory.java:56)
	at com.amazonaws.http.apache.client.impl.ApacheHttpClientFactory.create(ApacheHttpClientFactory.java:38)
	at com.amazonaws.http.AmazonHttpClient.<init>(AmazonHttpClient.java:347)
	at com.amazonaws.http.AmazonHttpClient.<init>(AmazonHttpClient.java:331)
	at com.amazonaws.AmazonWebServiceClient.<init>(AmazonWebServiceClient.java:239)
	at com.amazonaws.AmazonWebServiceClient.<init>(AmazonWebServiceClient.java:225)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.<init>(AWSSecurityTokenServiceClient.java:298)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.<init>(AWSSecurityTokenServiceClient.java:284)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder.build(AWSSecurityTokenServiceClientBuilder.java:61)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder.build(AWSSecurityTokenServiceClientBuilder.java:27)
	at com.amazonaws.client.builder.AwsSyncClientBuilder.build(AwsSyncClientBuilder.java:46)
	at software.amazon.msk.auth.iam.internals.MSKCredentialProvider$ProviderBuilder.createSTSRoleCredentialProvider(MSKCredentialProvider.java:316)
	at software.amazon.msk.auth.iam.internals.MSKCredentialProvider$ProviderBuilder.lambda$getStsRoleProvider$4(MSKCredentialProvider.java:308)
	at java.base@17.0.8/java.util.Optional.map(Optional.java:260)
	at software.amazon.msk.auth.iam.internals.MSKCredentialProvider$ProviderBuilder.getStsRoleProvider(MSKCredentialProvider.java:284)
	at software.amazon.msk.auth.iam.internals.MSKCredentialProvider$ProviderBuilder.getProviders(MSKCredentialProvider.java:246)
	at software.amazon.msk.auth.iam.internals.MSKCredentialProvider.<init>(MSKCredentialProvider.java:103)
	at software.amazon.msk.auth.iam.internals.MSKCredentialProvider.<init>(MSKCredentialProvider.java:99)
	at software.amazon.msk.auth.iam.IAMClientCallbackHandler.lambda$configure$1(IAMClientCallbackHandler.java:54)
	at java.base@17.0.8/java.util.Optional.map(Optional.java:260)
	at software.amazon.msk.auth.iam.IAMClientCallbackHandler.configure(IAMClientCallbackHandler.java:54)
	at org.apache.kafka.common.network.SaslChannelBuilder.configure(SaslChannelBuilder.java:151)
	... 22 more
```

**To Reproduce**
Here is configuration to connect to MSK:
```json
{
  "currentContext" : "dev",
  "dev" : {
    "configFile" : null,
    "configProps" : {
      "kafka.client.bootstrap.servers" : <OMITTED>,
      "kafka.client.security.protocol" : "SASL_SSL",
      "kafka.client.sasl.mechanism" : "AWS_MSK_IAM",
      "kafka.client.sasl.jaas.config" : "software.amazon.msk.auth.iam.IAMLoginModule required awsRoleArn=\"<OMITTED>\";",
      "kafka.client.sasl.client.callback.handler.class" : "software.amazon.msk.auth.iam.IAMClientCallbackHandler"
      
    }
  }
}
```
*Note the `awsRoleArn`*

execute `jikkou health get kafkabroker`

**Expected behavior**
I expect the same behavior as a non-native Java project: that is, AWS SDK can authenticate with MSK via IAM by assuming the given role.

**Runtime environment**
 - OS: MacOS 13.6
 - Jikkou: 0.30
 - Kafka Cluster Version: 2.8.2 (MSK)

**Additional context**
Here's the solution: https://github.com/oracle/graal/issues/1028

**Comment by fhussonnois** [comment]
Hi @singhbaljit, thank you for reporting this issue. The problem should be currenly fixed in the early-access release. The fix will be included in the next coming release.

**Comment by fhussonnois** [comment]
Fixed in 0.32.0
